tests/functional/conftest.py

```diff
@@ -94,7 +94,17 @@ def reset_gitlab(gl: gitlab.Gitlab) -> None:
             helpers.safe_delete(deploy_token)
         logging.info(f"Deleting project: {project.path_with_namespace!r}")
         helpers.safe_delete(project)
+
     for group in gl.groups.list():
+
+        # skip deletion of a descendant group to prevent scenarios where parent group gets deleted leaving a dangling descendant whose deletion will throw 404s.
+        # descendant groups cannot contain the `/` special character: https://docs.gitlab.com/ee/user/reserved_names.html#limitations-on-project-and-group-names
+        if "/" in group.full_path:
```

Comment:
This might be simpler, check if it has a parent:
```suggestion
        if group.parent_id:
```